executor/create_table_insert.go

```diff
@@ -0,0 +1,200 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package executor
+
+import (
+	"github.com/pingcap/errors"
+	"github.com/pingcap/parser/ast"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/table/tables"
+	"github.com/pingcap/tidb/tablecodec"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/chunk"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/net/context"
+)
+
+// CreateTableInsertExec represents an insert executor when creating table, it is basically similar to `InsertExec` except:
+//  1. It is instantiated and executed at DDL owner server, during the execution of 'create table' DDL job
+//  2. The 'on duplicate' behavior(ERROR/IGNORE/REPLACE) is specified by 'onDuplicate' option
+// See 'https://dev.mysql.com/doc/refman/5.7/en/create-table-select.html' for more details
+type CreateTableInsertExec struct {
```

Comment:
We already have `insert ignore` and `replace` statement, can we reuse them?
```sql
create table ... from select ...    ==> create table + insert into table (c1,c2...) select ...
create table ... ignore select ...  ==> create table + insert ignore into table (c1,c2...) select ... 
create table ... replace select ... ==> create table + replace into table (c1,c2...) select ... 

```

Reply:
I've removed most of the code of `CreateTableInsertExec` and replaced them with `InsertExec` and `ReplaceExec`. It's a very good idea, thank you!